ui/components/MessagesTable/MessagesTable.tsx

```diff
@@ -264,30 +268,42 @@ export function MessagesTable({
                           <Cell>
                             {row.attributes.key ? (
                               <>
-                              <UnknownValuePreview
-                                value={row.attributes.key}
-                                highlight={filterQuery}
-                                onClick={() => {
-                                  setDefaultTab("key");
-                                  onSelectMessage(row);
-                                }}
-                              />
-                              {row.relationships.keySchema && (
-                                <TextContent>
-                                  <Text component={"small"}>
-                                    {row.relationships.keySchema?.meta?.artifactType && (
-                                        <>
-                                        {row.relationships.keySchema?.meta?.artifactType}
-                                        </>
-                                    )}
-                                    {row.relationships.keySchema?.meta?.errors && (
+                                <UnknownValuePreview
+                                  value={row.attributes.key}
+                                  highlight={filterQuery}
+                                  onClick={() => {
+                                    setDefaultTab("key");
+                                    onSelectMessage(row);
+                                  }}
+                                />
+                                {row.relationships.keySchema && (
+                                  <TextContent>
+                                    <Text component={"small"}>
+                                      {row.relationships.keySchema?.meta
+                                        ?.name && (
+                                        <ExternalLink
+                                          testId={"key-schema"}
+                                          href={`${baseurl}/${row.relationships.keySchema?.links?.content}`}
+                                        >
```

Comment:
It seems a little funky to nest the schema under the topic messages, but I realize that there aren't really any alternatives given what the API is returning for the link. Maybe in a future PR the API can return the registry name and schema ID as separate fields and the pages no longer need to be nested this way.